spec.md

```diff
@@ -0,0 +1,948 @@
+
+In absence of the `VERSION` header implementations SHOULD assume the version 0.3.0.
+Implementations SHOULD reject a file based on the value of the `VERSION` header,
+in particular if the value is syntactically invalid.
+Applications MAY reject prerelease-versions altogether.
+
+The `VERSION` header SHOULD be the first header in a file.
+
+### 3.4. The `BPM` Header
+
+```
+Required:     Yes
+Multi-Valued: No
+Syntax:       1*DIGIT [ period 1*DIGIT ]
+Since:        0.1.0
+```
+
+The `BPM` header indicates the number of beats per minute.
+The notes in a song are quantized in beats.
+A single beat is the smallest unit of time that can be present in a song.
+The value is a 32 bit floating point number.
+The value of this tag is arbitrary in the sense that it is usually 4 to 8 times higher than the actual BPM of a song.
+
+> [!WARNING]
+>
+> **Breaking Change** in version 2.0.0
+>
+> In versions 0.x and 1.x the value of the `BPM` header was implicitly quadrupled.
+> This is **not** the case for version 2.0.0 and above.
+
+> [!CAUTION]
+>
+> The removal of the implicit quadrupling has not been decided yet.
+
+> [!WARNING]
+>
+> **Breaking Change** in version 2.0.0
+>
+> In versions 0.x and 1.x the comma was allowed as decimal separator in addition to the period.
+> This feature has been removed in version 2.0.0 and above.
+
+> [!CAUTION]
+>
+> The removal of the comma as a valid decimal separator has not been decided yet.
+
+### 3.5. The `AUDIO` Header
+
+```
+Required:     Yes
+Multi-Valued: No
+Since:        1.1.0
+```
+
+The `AUDIO` header contains a file reference (as defined in section 3.2.) to an audio file.
+This file contains the full version of a song (including instrumentals and vocals).
+Supported audio formats are an implementation detail.
+
+> [!CAUTION]
+>
+> If media files should follow a specific naming scheme has not been decided yet.
+
+### 3.6. The `MP3` Header
+
+```
+Required:     No (Version 2.x), Yes (Versions 0.x and 1.x)
+Multi-Valued: No
+Since:        0.1.0
+Deprecated:   1.1.0
+Removed:      2.0.0
+```
+
+The `MP3` header contains a file reference (as defined in section 3.2.) to an audio file.
+This header has been replaced by the `AUDIO` header.
+Implementations MUST disregard the `MP3` header if an `AUDIO` header is present (even if the specified file cannot be found or processed).
+
+### 3.7. The `COVER`, `BACKGROUND`, and `VIDEO` Headers
+
+```
+Required:     No
+Multi-Valued: No
+Since:        0.2.0
+```
+
+The headers `COVER`, `BACKGROUND`, and `VIDEO` contain file references to image files or in case of `VIDEO` video files.
+Implementations MAY use these files to display cover artwork and background graphics during gameplay.
+Supported image and video formats are an implementation detail.
+
+For the best compatibility `COVER` images should be square and be no larger than 1920 pixels in width and height.
```

Comment:
I agree that the image size should be an implementation detail. I originally included it, because it is mentioned on the [website](https://usdx.eu/format/#cover) but would have moved to remove it in a follow-up issue. I removed the paragraph.

I personally wouldn't limit the use of these headers in the spec if they are not vital to the core aspect of singing karaoke. Would you change the wording here? To me the MAY imperative makes it clear enough that implementations can choose to ignore these headers as they see fit.